### PR TITLE
Hive is not getting closed , as it picks up HiveDB from ThreadLocal. …

### PR DESCRIPTION
**Pull request by rishitesh**
## Changes proposed in this pull request
  Hive is not getting closed , as it picks up HiveDB from ThreadLocal. SparkListener is a separate thread and it does not clean up the current connection to metastore.
If application starts stops SparkContext then this results in CacheClosedException from stale connections.

Hence added clean-up code while creating a new client. 

## Patch testing

pre-checkin ran successfully

## ReleaseNotes.txt changes
NA

## Other PRs 
NA

### CODE REVIEW

**Comment by kneeraj** [comment]
Looks good.

File: core/src/main/scala/org/apache/spark/sql/hive/SnappyStoreHiveCatalog.scala [hunk 1]
```diff
@@ -145,6 +145,8 @@ class SnappyStoreHiveCatalog(externalCatalog: ExternalCatalog,
    */
   protected def configure(): Map[String, String] = Map.empty
 
+
+
```
Comment (sumwale) [inline core/src/main/scala/org/apache/spark/sql/hive/SnappyStoreHiveCatalog.scala]: unnecessary whitespace added

**Comment by sumwale** [comment]
looks good
